python/tests/test_stream.py

```diff
@@ -80,3 +80,23 @@ def test_response_exception():
     resp.flush()
     with pytest.raises(TChannelException):
         yield resp.write_body("aaaa")
+
+
+@pytest.mark.gen_test
+def test_error_during_stream(io_loop):
+    stream = InMemStream()
+
+    @pytest.mark.gen_test
```

Comment:
This shouldn't be another `gen_test`. It would be a `coroutine`, but if you `return stream.set_exception(e)` (which should be a future) then you don't need a decorator at all.